prov/efa/src/rdm/rxr_msg.c

```diff
@@ -283,6 +252,34 @@ static
 ssize_t rxr_msg_sendmsg(struct fid_ep *ep, const struct fi_msg *msg,
 			uint64_t flags)
 {
+	struct efa_rdm_peer *peer;
+	struct rxr_ep *rxr_ep;
+	void *shm_desc[RXR_IOV_LIMIT] = {NULL};
+	struct fi_msg *shm_msg;
+	void **efa_desc = NULL;
+	fi_addr_t efa_addr;
+	int ret;
+
+	rxr_ep = container_of(ep, struct rxr_ep, base_ep.util_ep.ep_fid.fid);
+	peer = rxr_ep_get_peer(rxr_ep, msg->addr);
```

Comment:
Come to think about it, now that we have peer api, we can do a separate refactoring to encapsulate the peer ops in rdm peer.

The code as-is is rather verbose and repetitive. Looking at the change stat, when the logic was in generic send, we were removing 100+ LOC overall. After we scattered the logic in each send api we are adding 100+ LOC.